File: etc/perl/AddExample.pl
# This example shows how new documents can be added.
# Documentation: http://basex.org/api
#
# (C) BaseX Team 2005-11, ISC License

use BaseXClient;
use warnings;
use strict;

eval {

  # create session
  my $session = Session->new("localhost", 1984, "admin", "admin");

  # create empty database
  $session->execute("create db database");
  print "\n".$session->info();
  
  # add document
  $session->add("World.xml", "/world", "<x>Hello World!</x>");
  print "\n".$session->info()."\n";
  
  # add document
  $session->add("Universe.xml", "", "<x>Hello Universe!</x>");
  print "\n".$session->info()."\n";
  
  # run query on database
  print "\n".$session->execute("xquery collection('database')")."\n";
  
  # drop database
  $session->execute("drop db database");
  
  # close session
  $session->close();
};

# print exception
print $@ if $@;

File: src/main/java/org/basex/examples/api/AddExample.java
package org.basex.examples.api;
import java.io.*;

/**
 * This example shows how documents can be added to databases, and how
 * existing documents can be replaced.
 *
 * Documentation: http://docs.basex.org/wiki/Clients
 *
 * @author BaseX Team 2005-12, BSD License
 */
public final class AddExample {
  /** Hidden default constructor. */
  private AddExample() { }

  /**
   * Main method.
   * @param args command-line arguments
   * @throws IOException I/O exception
   */
  public static void main(final String[] args) throws IOException {
    // create session
    final BaseXClient session = new BaseXClient("localhost", 1984, "admin", "admin");

    try {
      // create empty database
      session.execute("create db database");
      System.out.println(session.info());

      // define input stream
      InputStream bais =
        new ByteArrayInputStream("<x>Hello World!</x>".getBytes());

      // add document
      session.add("world/world.xml", bais);
      System.out.println(session.info());

      // define input stream
      bais = new ByteArrayInputStream("<x>Hello Universe!</x>".getBytes());

      // add document
      session.add("universe.xml", bais);
      System.out.println(session.info());

      // run query on database
      System.out.println(session.execute("xquery collection('database')"));

      // define input stream
      bais = new ByteArrayInputStream("<x>Hello Replacement!</x>".getBytes());

      // add document
      session.replace("universe.xml", bais);
      System.out.println(session.info());

      // run query on database
      System.out.println(session.execute("xquery collection('database')"));

      // drop database
      session.execute("drop db database");

    } finally {
      // close session
      session.close();
    }
  }
}
